### ISSUE TITLE
`magit-diff` yields "transient-setup: Wrong type argument: integer-or-marker-p, nil"

### ISSUE THREAD

**Issue by peterbecich** [magit/magit#4881]
I have uninstalled/re-installed Magit and its dependencies as instructed, and have also deleted all dependencies in `~/.emacs.d/elpa`, installed only `magit` and its direct dependencies, including `transient`; issue persists.

Magit versions tested:
`Magit 20230217.2343 [>= 3.3.0.50-git], Git 2.38.1, Emacs 28.2, gnu/linux`
and
`3.3.0`

Transient version: `20230218.2130`

In the Magit Overview, `M-x magit-diff` produces:

> transient-setup: Wrong type argument: integer-or-marker-p, nil

The shortcut `d` produces the same output.

Usually `magit-diff` asks for a branch to compare to, and then opens a buffer with the diff.
Here is the backtrace:
```
Debugger entered--Lisp error: (wrong-type-argument integer-or-marker-p nil)
  transient--exit-and-debug(error (wrong-type-argument integer-or-marker-p nil))
  read-kbd-macro(nil)
  kbd(nil)
  transient--make-transient-map()
  transient-setup(magit-diff)
  magit-diff()
  funcall-interactively(magit-diff)
  call-interactively(magit-diff nil nil)
  command-execute(magit-diff)
```

Thanks very much for Magit, it is phenomenal

**Comment by pjj** [comment]
I'm getting the same error.  I also get the same error with several other magit commands, for example magit-log (l) or magit-commit (c).  I'm using TOT Emacs (750bc57cbb8).

**Comment by peterbecich** [comment]
I reverted to https://github.com/magit/magit/commit/467bb21f7d773433a50ac6a89087b6c2b41ab86d (manual install) and it works again. I believe this also provides the `transient` dependency; `transient` and `magit` are not included in the `*Packages*` list now.

**Comment by BrettWitty** [comment]
> I'm getting a similar error without the "transient-setup: ". I also get that error with several other magit commands, for example magit-log (l) or magit-commit (c). I'm using TOT Emacs (750bc57cbb8) with the latest magit/transient. The explicit subcommands work fine, e.g. magit-commit-create, or magit-log-all.

I'm getting the same with melpa magit version `20230217.2343`.

**Comment by tarsius** [comment]
I've reverted the offending commit in `transient`

Unfortunately updating only `transient` is not enough to remove the bug. The bug is in code that the `transient-define-prefix` macro expands to. So after updating `transient`, you also have to recompile `magit` and all other packages that depend on `transient`, while the faulty version of the macro was in use.

**Comment by peterbecich** [comment]
Thank you, it is fixed on Magit `20230219.1247`
